### PR TITLE
Message Update

### PR DESCRIPTION
**Pull request by donkkill13**
I added reaction based things to the Message part, added WIP events for Message Reactions, and added some functions to the Message part.

### CODE REVIEW

**Comment by sciencefyll** [comment]
MERGE IT

**Comment by uniquoooo** [comment]
Looks good, I will clean this up a bit and merge later today hopefully.

File: src/Discord/Parts/Channel/Message.php [hunk 1]
```diff
@@ -79,6 +80,125 @@ public function reply($text)
...
+     * @return \React\Promise\Promise
+     */
+    public function react($emoticon)
+    {
+        $deferred = new Deferred();
+
+        $this->http->put(
+            "channels/{$this->channel->id}/messages/{$this->id}/reactions/{$emoticon}/@me"
+        )->then(
+            function ($response) use ($deferred) {
```
Comment (uniquoooo) [inline src/Discord/Parts/Channel/Message.php]: Changing this to match the rest of the library with `\React\Partial\bind_right($this->resolve, $deferred)`.

File: src/Discord/Parts/Channel/Message.php [hunk 2]
```diff
@@ -79,6 +80,125 @@ public function reply($text)
...
+    }
+    
+    /**
+     * Delete a reaction.
+     *
+     * @param array $settings  (id and emoticon)
+     *
+     * @return \React\Promise\Promise
+     */
+    public function react_delete($type, $emoticon = null, $id = null)
```
Comment (uniquoooo) [inline src/Discord/Parts/Channel/Message.php]: I will change this to `deleteReaction` to match the function name style of the rest of the library. Also changing the `types` to constants.

File: src/Discord/Parts/Channel/Message.php [hunk 3]
```diff
@@ -79,6 +80,125 @@ public function reply($text)
...
+    }
+    
+    /**
+     * Updates the message.
+     *
+     * @param string $text  The text to send in the message.
+     *
+     * @return \React\Promise\Promise
+     */
+    public function edit($text)
```
Comment (uniquoooo) [inline src/Discord/Parts/Channel/Message.php]: I'd rather keep editing and deleting to the message repositories so I will remove the `edit` and `delete` functions.

**Comment by uniquoooo** [comment]
Thanks!

**Pull request merged**